### PR TITLE
Work around for project dropdown overflows

### PR DESCRIPTION
**Pull request by b12f**
The new hierarchical project dropdowns caused an issue because long
project names are cut off by default. This makes it impossible to
distinguish when the naming conventions puts the difference between
project at the end of their name, for example:

"Project with long name 1" and "Project with long name 2" will both
show up as "Project with...".

* This PR adds a `title` attribute with the name, so hovering the
option will give users the possibility of reading the full name.

* It also widens the result list in a rather hacky way. It works, but
could cause the result list to be cut off by the screen edge if
the project select is e.g. all the way on the right in the wp table.

* Further it takes care the input does not overflow the table cell ( see #42939)

Closes
https://community.openproject.org/projects/openproject/work_packages/43083/activity
https://community.openproject.org/projects/openproject/work_packages/42939/activity

### CODE REVIEW

**Review by HDinger — CHANGES_REQUESTED** [review]
I don't really like the Hack with the extra wrapper div which will be rendered everywhere. And I think it is not necessary. You can still append to the body and just pass a modifier class through to the ng-select and add the min-width style to that. The `op-autocompleter` already supports additional classes, you'd just have to add them the `project-autocompleter`

**Review by HDinger — COMMENTED** [review]
I still don't like it, but because of the discussed reasons I don't see another solution either. Please add some explanatory comments and link the GitHub issues. After that it is good to merge from my side.

**Review by HDinger — APPROVED** [review]
The changes are fine for me but since I pushed some commits for the second mentioned ticket, I'd like to have a third opinion (cc @Kharonus)

File: frontend/src/global_styles/common/wide-autocomplete-wrapper/wide-autocomplete-wrapper.sass
Comment (Kharonus) [inline frontend/src/global_styles/common/wide-autocomplete-wrapper/wide-autocomplete-wrapper.sass:26]: ouf, I see, what @HDinger meintioned about the hack. I do not know all the reasons, but the comments are sufficient to mark this as a "dirty hack". I think, we can't do anything right now, to be careful and observe, if this breaks something on other places in code.

File: frontend/src/app/shared/components/autocompleter/project-autocompleter/project-autocompleter.component.ts
Comment (Kharonus) [inline frontend/src/app/shared/components/autocompleter/project-autocompleter/project-autocompleter.component.ts:120]: what is the advantage of using an input in combination with a hostbinding, instead of simply using `ngClass` in whatever template this component is used? IMHO it kinda hides the effect, when simply looking on the parent template.

**Review by Kharonus — APPROVED** [review]
I'd approve those changes, if there is a good reason/answer to that question. ;)

File: frontend/src/app/shared/components/autocompleter/project-autocompleter/project-autocompleter.component.ts
Comment (HDinger) [reply frontend/src/app/shared/components/autocompleter/project-autocompleter/project-autocompleter.component.ts:120]: In my eyes it has two main advantages:

1. Using an input is kind of a documentation the inside the belonging component. If you'd only use a class in the parent component, you'd not see its effect in the code. You'd have to go to the component style sheet to check it which normally nobody does, tbh. 
2. In general using an `ngClass` is very error-prone to spelling mistakes. As soon as someone uses the wrong class, it won't work and nobody would notice because there is no compile error.

Comment (Kharonus) [reply frontend/src/app/shared/components/autocompleter/project-autocompleter/project-autocompleter.component.ts:120]: agreed, approved it is then

**Pull request merged**
